packages/react-dom/src/events/__tests__/SyntheticEvent-test.js

```diff
@@ -9,114 +9,234 @@
+    var event;
+    event = document.createEvent('Event');
+    event.initEvent('click', true, true);
+    node.dispatchEvent(event);
 
-    expect(nativeEvent.returnValue).toBe(false);
+    expect(expectedCount).toBe(1);
   });
 
   it('should be prevented if nativeEvent is prevented', () => {
-    expect(createEvent({defaultPrevented: true}).isDefaultPrevented()).toBe(
-      true,
-    );
-    expect(createEvent({returnValue: false}).isDefaultPrevented()).toBe(true);
+    var node;
+    var expectedCount = 0;
+
+    var eventHandler = syntheticEvent => {
+      expect(syntheticEvent.isDefaultPrevented()).toBe(true);
+
+      expectedCount++;
+    };
+    node = ReactDOM.render(<div onClick={eventHandler} />, container);
+
+    var event;
+    event = document.createEvent('Event');
+    event.initEvent('click', true, true);
+    Object.defineProperty(event, 'defaultPrevented', {
+      get() {
+        return true;
+      },
+    });
+    node.dispatchEvent(event);
+
+    event = document.createEvent('Event');
+    event.initEvent('click', true, true);
+    // Emulate IE8
+    Object.defineProperty(event, 'defaultPrevented', {
+      get() {},
+    });
+    Object.defineProperty(event, 'returnValue', {
+      get() {
+        return false;
+      },
+    });
+    node.dispatchEvent(event);
+
+    expect(expectedCount).toBe(2);
   });
 
   it('should be able to `stopPropagation`', () => {
-    var nativeEvent = {};
-    var syntheticEvent = createEvent(nativeEvent);
+    var node;
+    var expectedCount = 0;
 
-    expect(syntheticEvent.isPropagationStopped()).toBe(false);
-    syntheticEvent.stopPropagation();
-    expect(syntheticEvent.isPropagationStopped()).toBe(true);
+    var eventHandler = syntheticEvent => {
+      expect(syntheticEvent.isPropagationStopped()).toBe(false);
+      syntheticEvent.stopPropagation();
+      expect(syntheticEvent.isPropagationStopped()).toBe(true);
 
-    expect(nativeEvent.cancelBubble).toBe(true);
+      expectedCount++;
+    };
+    node = ReactDOM.render(<div onClick={eventHandler} />, container);
+
+    var event;
+    event = document.createEvent('Event');
+    event.initEvent('click', true, true);
+    node.dispatchEvent(event);
+
+    expect(expectedCount).toBe(1);
   });
 
   it('should be able to `persist`', () => {
-    var syntheticEvent = createEvent({});
+    var node;
+    var expectedCount = 0;
+
+    var eventHandler = syntheticEvent => {
+      expect(syntheticEvent.isPersistent()).toBe(false);
+      syntheticEvent.persist();
+      expect(syntheticEvent.isPersistent()).toBe(true);
 
-    expect(syntheticEvent.isPersistent()).toBe(false);
-    syntheticEvent.persist();
-    expect(syntheticEvent.isPersistent()).toBe(true);
+      expectedCount++;
+    };
+    node = ReactDOM.render(<div onClick={eventHandler} />, container);
+
+    var event;
+    event = document.createEvent('Event');
+    event.initEvent('click', true, true);
+    node.dispatchEvent(event);
+
```

Comment:
It would be nice to test that we can still access `syntheticEvent` fields here. Since that's the thing that `persist` allows.

Let's save `syntheticEvent` in a variable and asset its `target` and `type` here.